boards/arm/waveshare_open103z/waveshare_open103z_defconfig

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_STM32F1X=y
+CONFIG_SOC_STM32F103XE=y
+# 72MHz system clock
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=72000000
+
+# enable uart driver
+CONFIG_SERIAL=y
+# enable console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# enable pinmux
+CONFIG_PINMUX=y
+
+# enable GPIO
+CONFIG_GPIO=y
+
+# enable SPI
+CONFIG_SPI=y
+CONFIG_SPI_STM32=y
+CONFIG_SPI_STM32_USE_HW_SS=y
```

Comment:
Peripheral activation (other than basics gpio, serial) is not part of default board configuration, this is application responsibility.
Also, the Kconfig system will take care of activating SPI_STM32 based on SPI
These should be removed.